spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1474,15 +1481,14 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
             None
           }
 
-        case Abs(child, _) =>
+        case Abs(child, failOnErr) =>
```

Comment:
since we are already using failOnErr, can we simply use this boolean instead of evalmode struct? what are you thoughts?

Reply:
Hi, thanks for your review! The intention why it is done this way is that in the Rust code the ansi mode is always treated the same way whether the expression supports the three ansi modes or only two. If not, we have to do a different treatment for both cases.